libvast/src/system/partition.cpp

```diff
@@ -98,109 +173,85 @@ caf::actor fetch_indexer(table_indexer& tbl, const data_extractor& dx,
       return [=](const curried_predicate&) { return row_ids; };
     });
   }
   if (ex.attr == system::timestamp_atom::value) {
     if (!caf::holds_alternative<timestamp>(x)) {
-      VAST_WARNING(tbl.state().self,
-                   "expected a timestamp as time extractor attribute , got:",
-                   x);
+      VAST_WARNING(
+        state_->self,
+        "expected a timestamp as time extractor attribute , got:", x);
       return nullptr;
     }
     // Find the column with attribute 'time'.
     auto pred = [](auto& x) {
       return caf::holds_alternative<time_type>(x.type)
              && has_attribute(x.type, "timestamp");
     };
+    auto layout = combined_type();
     auto& fs = layout.fields;
     auto i = std::find_if(fs.begin(), fs.end(), pred);
     if (i == fs.end())
       return nullptr;
     // Redirect to "ordinary data lookup".
     auto pos = static_cast<size_t>(std::distance(fs.begin(), i));
     data_extractor dx{layout, vast::offset{pos}};
-    return fetch_indexer(tbl, dx, op, x);
+    return fetch_indexer(dx, op, x);
   }
-  VAST_WARNING(tbl.state().self, "got unsupported attribute:", ex.attr);
+  VAST_WARNING(state_->self, "got unsupported attribute:", ex.attr);
   return nullptr;
 }
 
-} // namespace
-
-evaluation_map partition::eval(const expression& expr) {
-  evaluation_map result;
-  // Step #1: use the expression to select matching layouts.
-  for (auto layout : layouts()) {
-    // Step #2: Split the resolved expression into its predicates and select
-    // all matching INDEXER actors per predicate.
-    auto resolved = resolve(expr, layout);
-    // Skip any layout that we cannot resolve.
-    if (resolved.empty())
-      continue;
-    // Add triples (offset, curried predicate, and INDEXER) to evaluation map.
-    evaluation_map::mapped_type triples;
-    for (auto& kvp: resolved) {
-      auto& pred = kvp.second;
-      auto get_indexer_handle = [&](const auto& ext, const data& x) {
-        if (auto i = get_or_add(layout))
-          return fetch_indexer(i->first, ext, pred.op, x);
-        VAST_ERROR(state_->self,
-                   "failed to initialize table_indexer for layout", layout,
-                   "-> query will not execute on the full data set");
-        return caf::actor{};
-      };
-      auto v = detail::overload(
-        [&](const attribute_extractor& ex, const data& x) {
-          return get_indexer_handle(ex, x); // clang-format fix
-        },
-        [&](const data_extractor& dx, const data& x) {
-          return get_indexer_handle(dx, x);
-        },
-        [](const auto&, const auto&) {
-          return caf::actor{}; // clang-format fix
-        });
-      auto hdl = caf::visit(v, pred.lhs, pred.rhs);
-      if (hdl != nullptr) {
-        triples.emplace_back(kvp.first, curried(pred), std::move(hdl));
-      }
+evaluation_triples partition::eval(const expression& expr) {
+  evaluation_triples result;
+  // Step #1: Pretend the partition is a table.
```

Comment:
I was looking for `Step #2` somewhere. Does it exist? Otherwise the comment is misleading.

Reply:
The comment has been refined.